CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JMultiChoiceComboBoxMenuNGTest.java

```diff
@@ -181,6 +185,72 @@ public void testaddButtonActionListener() {
         assertFalse(spyMenu.isVisible());
     }
 
+    /**
+     * Test of addMenuItemActionListener method, of class
+     * JMultiChoiceComboBoxMenu when the selectedItem contains.
+     */
+    @Test
+    public void testAddMenuItemActionListener_selectedItem_contains() {
+        System.out.println("addMenuItemActionListener_selectedItem_contains");
+
+        final JMultiChoiceComboBoxMenu instance = new JMultiChoiceComboBoxMenu("Text", items);
+        final ActionEvent actionEventMock = mock(ActionEvent.class);
+        final Map<JMenuItem, String> menuItems = instance.getMenuItems();
+        final String itemSelected = "Item 3";
+        final JMenuItem manuItem = getKey(menuItems, itemSelected);
+        final ListSelectionListener listSelectionListenerMock1 = mock(ListSelectionListener.class);
+
+        doReturn(manuItem).when(actionEventMock).getSource();
+
+        instance.addSelectionListener(listSelectionListenerMock1);
+        instance.setSelectedItem(itemSelected);
+
+        final Set<String> resultSet = instance.getSelectedItems();
+        assertEquals(resultSet.size(), 1);
+
+        manuItem.getActionListeners()[0].actionPerformed(actionEventMock);
+
+        verify(listSelectionListenerMock1, times(1)).valueChanged(any(ListSelectionEvent.class));
+        assertEquals(resultSet.size(), 0);
+    }
+
+    /**
+     * Test of addMenuItemActionListener method, of class
+     * JMultiChoiceComboBoxMenu when the selectedItem does not contain.
+     */
+    @Test
+    public void testAddMenuItemActionListener_selectedItem_does_not_contain() {
+        System.out.println("addMenuItemActionListener_selectedItem_does_not_contain");
+
+        final JMultiChoiceComboBoxMenu instance = new JMultiChoiceComboBoxMenu("Text", items);
+        final ActionEvent actionEventMock = mock(ActionEvent.class);
+        final Map<JMenuItem, String> menuItems = instance.getMenuItems();
+        final String itemSelected = "Item 1";
+        final JMenuItem manuItem = getKey(menuItems, "Item 4");
+        final ListSelectionListener listSelectionListenerMock1 = mock(ListSelectionListener.class);
+
+        doReturn(manuItem).when(actionEventMock).getSource();
+
+        instance.addSelectionListener(listSelectionListenerMock1);
+        instance.setSelectedItem(itemSelected);
+
+        final Set<String> resultSet = instance.getSelectedItems();
+        assertEquals(resultSet.size(), 1);
+
+        manuItem.getActionListeners()[0].actionPerformed(actionEventMock);
+
+        verify(listSelectionListenerMock1, times(1)).valueChanged(any(ListSelectionEvent.class));
+        assertEquals(resultSet.size(), 2);
+    }
+
+    public <K, V> K getKey(Map<K, V> map, V value) {
+        for (Entry<K, V> entry : map.entrySet()) {
+            if (entry.getValue().equals(value)) {
+                return entry.getKey();
+            }
+        }
+        return null;
```

Comment:
I would make this private since this won't be used outside the test class. 

Probably worth noting though that while it may be ok for the test class, this function does seem to somewhat assume that values are unique (which isn't the case with maps) which means in the general case it could return a different result each time depending on how the map is iterated through

Reply:
Yes agree, added the support of multiple values for clarity, although the test cases this function is used only contain unique values.